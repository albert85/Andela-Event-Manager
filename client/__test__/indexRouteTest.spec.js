import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import AllRoutes from '../src/routes/routes';

let props;
let isComponentMounted;

configure({ adapter: new Adapter() });


const getComponentStatus = () => {
  if (!isComponentMounted) {
    isComponentMounted = shallow(<AllRoutes {...props} />);
  }
  return isComponentMounted;
};

describe('Index Routes', () => {
  it('should successfully rendered route index', () => {
    expect(getComponentStatus()).toMatchSnapshot();
  });
});
