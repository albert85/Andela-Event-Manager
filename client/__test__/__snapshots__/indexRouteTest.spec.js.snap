// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Routes should successfully rendered route index 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/signup"
        />
        <Route
          component={[Function]}
          path="/event-home-page"
        />
        <Route
          component={[Function]}
          path="/centers"
        />
        <Route
          component={[Function]}
          path="/booking-details"
        />
        <Route
          component={[Function]}
          path="/center-details"
        />
        <Route
          component={[Function]}
          path="/edit-center-details"
        />
        <Route
          component={[Function]}
          path="/edit-event"
        />
        <Redirect
          push={false}
          to="/"
        />
      </Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            path="/signup"
          />,
          <Route
            component={[Function]}
            path="/event-home-page"
          />,
          <Route
            component={[Function]}
            path="/centers"
          />,
          <Route
            component={[Function]}
            path="/booking-details"
          />,
          <Route
            component={[Function]}
            path="/center-details"
          />,
          <Route
            component={[Function]}
            path="/edit-center-details"
          />,
          <Route
            component={[Function]}
            path="/edit-event"
          />,
          <Redirect
            push={false}
            to="/"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/event-home-page",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/centers",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/booking-details",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/center-details",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/edit-center-details",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/edit-event",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/signup"
          />
          <Route
            component={[Function]}
            path="/event-home-page"
          />
          <Route
            component={[Function]}
            path="/centers"
          />
          <Route
            component={[Function]}
            path="/booking-details"
          />
          <Route
            component={[Function]}
            path="/center-details"
          />
          <Route
            component={[Function]}
            path="/edit-center-details"
          />
          <Route
            component={[Function]}
            path="/edit-event"
          />
          <Redirect
            push={false}
            to="/"
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/signup"
            />,
            <Route
              component={[Function]}
              path="/event-home-page"
            />,
            <Route
              component={[Function]}
              path="/centers"
            />,
            <Route
              component={[Function]}
              path="/booking-details"
            />,
            <Route
              component={[Function]}
              path="/center-details"
            />,
            <Route
              component={[Function]}
              path="/edit-center-details"
            />,
            <Route
              component={[Function]}
              path="/edit-event"
            />,
            <Redirect
              push={false}
              to="/"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/event-home-page",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/centers",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/booking-details",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/center-details",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/edit-center-details",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/edit-event",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "push": false,
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
